pkg/webui/console/views/device-add/repository/repository.js

```diff
@@ -189,54 +192,59 @@ const DeviceRepository = props => {
-          onBrandChange={handleBrandChange}
-          onModelChange={handleModelChange}
-          onHwVersionChange={handleHwVersionChange}
-          onFwVersionChange={handleFwVersionChange}
-          onBandChange={handleBandChange}
-        />
-        {showProgressHint && (
-          <ProgressHint
-            manualLinkPath={`/applications/${appId}/devices/add/manual`}
-            supportLink={supportLink}
-          />
-        )}
-        {showOtherHint && (
-          <OtherHint
-            manualLinkPath={`/applications/${appId}/devices/add/manual`}
-            manualGuideDocsPath="/devices/adding-devices/"
-          />
-        )}
-        <hr className={style.hRule} />
-        <Form.SubTitle title={m.enterDataTitle} />
-        {showRegistrationForm ? (
-          <Registration
-            template={template}
-            fetching={templateFetching}
-            prefixes={prefixes}
-            mayEditKeys={mayEditKeys}
-          />
-        ) : (
-          <Message content={m.enterDataDescription} component="p" />
-        )}
-        <SubmitBar align="end">
-          <Form.Submit
-            message={m.register}
-            component={SubmitButton}
-            disabled={!showRegistrationForm}
-          />
-        </SubmitBar>
-      </Form>
+      <Row>
+        <Col>
+          <Form
+            formikRef={formRef}
+            onSubmit={handleSubmit}
+            initialValues={initialValues}
+            validationSchema={validationSchema}
+            validationContext={validationContext}
+            error={error}
+          >
+            <Form.SubTitle title={m.selectDeviceTitle} />
+            <Selection
+              onBrandChange={handleBrandChange}
+              onModelChange={handleModelChange}
+              onHwVersionChange={handleHwVersionChange}
+              onFwVersionChange={handleFwVersionChange}
+              onBandChange={handleBandChange}
+            />
+            {showProgressHint && (
+              <ProgressHint
+                manualLinkPath={`/applications/${appId}/devices/add/manual`}
+                supportLink={supportLink}
+              />
+            )}
+            {showOtherHint && (
+              <OtherHint
+                manualLinkPath={`/applications/${appId}/devices/add/manual`}
+                manualGuideDocsPath="/devices/adding-devices/"
+              />
+            )}
+            {showDeviceCard && <Card brandId={brand} modelId={model} template={template} />}
+            <hr className={style.hRule} />
+            <Form.SubTitle title={m.enterDataTitle} />
+            {showRegistrationForm ? (
+              <Registration
+                template={template}
+                fetching={templateFetching}
+                prefixes={prefixes}
+                mayEditKeys={mayEditKeys}
+              />
+            ) : (
+              <Message content={m.enterDataDescription} component="p" />
+            )}
+            <SubmitBar align="end">
```

Comment:
```suggestion
            <SubmitBar align="start">
```

Not related to this PR but needs to be changed.